mplayer: added some stuff to make OSD menu work via X11

ggm-at-apnic opened the pull request:
OSX El Capitan appears to have broken the non-X11 OSD. Several people are commenting on this online. I went into the Formula for mplayer, added back the x11 dependencies to enable x11 and turn on menu. This worked for me with XQuartz.

I put in a -with-x11OSD option to turn this on. I tried to follow the logic of how other people do things. 

mplayer uses a home-made configure which is *not* entirely normal, and needed multiple libraries passed in to the link phase. I used two flags to do this because the quoting to pass -lX11 -lXext didn't work on one line via the Formula. Thats a hack, but it worked for me.

apjanke commented:
Builds failed. Looks like the X11 dependency is not optional.

```
clang -MMD -MP -D_ISOC99_SOURCE -I. -Iffmpeg -O2 -march=corei7 -pipe -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -mdynamic-no-pic  -fpie -DPIC  -I/usr/X11/include -DZLIB_CONST -c -o mplayer.o mplayer.c
In file included from mplayer.c:133:
./libvo/x11_common.h:23:10: fatal error: 'X11/Xlib.h' file not found
#include <X11/Xlib.h>
         ^
1 error generated.
make: *** [mplayer.o] Error 1
```

It would be really nice if we could get this to build without the X11 dependency. We try to avoid depending on X11 in core formulae.

mikemcquaid commented on Library/Formula/mplayer.rb:
```diff
@@ -23,6 +23,9 @@ class Mplayer < Formula
     patch :DATA
   end
 
+  option "with-x11OSD", "Enable x11 for freefont/freetype menu support"
```
Just use `with-x11` for the option name here.

ggm-at-apnic replied:
done, tested, committed (to my fork) Thanks!

On 23 December 2015 at 18:19, Mike McQuaid <notifications@github.com> wrote:

> In Library/Formula/mplayer.rb
> <https://github.com/Homebrew/homebrew/pull/47287#discussion_r48328082>:
>
> > @@ -23,6 +23,9 @@ class Mplayer < Formula
> >      patch :DATA
> >    end
> >
> > +  option "with-x11OSD", "Enable x11 for freefont/freetype menu support"
>
> Just use with-x11 for the option name here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Homebrew/homebrew/pull/47287/files#r48328082>.
>

DomT4 commented on Library/Formula/mplayer.rb:
```diff
@@ -47,11 +50,15 @@ def install
 
     args << "--enable-caca" if build.with? "libcaca"
 
+    args << "--enable-x11" if build.with? "x11OSD"
```
You can stick these in a block...

```ruby
if build.with? "blah"
  args << "blah"
  args << "blah2"
  args << "blah3"
end
```
And such.

ggm-at-apnic replied:
Done. Thanks for the cluestick hit, the better way worked fine!

On 24 December 2015 at 05:36, Dominyk Tiller <notifications@github.com>
wrote:

> In Library/Formula/mplayer.rb
> <https://github.com/Homebrew/homebrew/pull/47287#discussion_r48373477>:
>
> >      ]
> >
> >      args << "--enable-caca" if build.with? "libcaca"
> >
> > +    args << "--enable-x11" if build.with? "x11OSD"
>
> You can stick these in a block...
>
> if build.with? "blah"
>   args << "blah"
>   args << "blah2"
>   args << "blah3"end
>
> And such.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Homebrew/homebrew/pull/47287/files#r48373477>.
>

ggm-at-apnic commented:
The whole *point* was to put the x11 dependency back in, and make menus
work. I absolutely agree it would be super nice not to do it, but libcaca
didn't work alone, and quite a lot of mplayer users out there are noting
that the OSD isn't working: the x11 dependency restores it.

On 23 December 2015 at 17:17, Andrew Janke <notifications@github.com> wrote:

> Builds failed. Looks like the X11 dependency is not optional.
>
> clang -MMD -MP -D_ISOC99_SOURCE -I. -Iffmpeg -O2 -march=corei7 -pipe -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -mdynamic-no-pic  -fpie -DPIC  -I/usr/X11/include -DZLIB_CONST -c -o mplayer.o mplayer.c
> In file included from mplayer.c:133:
> ./libvo/x11_common.h:23:10: fatal error: 'X11/Xlib.h' file not found
> #include <X11/Xlib.h>
>          ^
> 1 error generated.
> make: *** [mplayer.o] Error 1
>
> It would be really nice if we could get this to build without the X11
> dependency. We try to avoid depending on X11 in core formulae.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Homebrew/homebrew/pull/47287#issuecomment-166819526>.
>

mikemcquaid commented:
Fails to install without X11 now: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/38781/version=el_capitan/testReport/junit/brew-test-bot/el_capitan/install_mplayer/

bfontaine commented on Library/Formula/mplayer.rb:
```diff
@@ -47,11 +50,18 @@ def install
+      args << "--extra-libs=-lX11"
+      args << "--extra-libs-mplayer=-lXext"
+      args << "--enable-menu"
+    end
```
Shouldn’t we keep `--disable-x11` if this is not built with `x11`? E.g.:
```rb
else
  args << "--disable-x11"
end
```

DomT4 commented:
Closing as dead. Please do feel free to reopen if you wish to with the comments addressed. Thank you for your submission to Homebrew :heart:

ggm-at-apnic commented:
Sorry for delay, I've been on the road with work. The comment about disable-x11 made perfect sense and I have added this into the .rb -If you do not --enable-x11 then it returns to the default --disable-x11 and this is the default.

I updated this patch for mplayer 1.3.0 and confirmed the negative and positive case. It no longer drags in x11 if you don't ask for it.

I also totally "get" that you're keen to avoid x11 dependencies, the thing is: without this, on El Capitan mplayer simply doesn't *have* an OSD. This is an expedient to get back some basic behaviour.
